pkg/assembler/backends/ent/backend/hashequal.go

```diff
@@ -61,6 +62,18 @@ func (b *EntBackend) IngestHashEqual(ctx context.Context, artifact model.Artifac
 	return toModelHashEqual(record.Unwrap()), nil
 }
 
+func (b *EntBackend) IngestHashEquals(ctx context.Context, artifacts []*model.ArtifactInputSpec, otherArtifacts []*model.ArtifactInputSpec, hashEquals []*model.HashEqualInputSpec) ([]*model.HashEqual, error) {
+	var result []*model.HashEqual
+	for i := range hashEquals {
+		he, err := b.IngestHashEqual(ctx, *artifacts[i], *otherArtifacts[i], *hashEquals[i])
```

Comment:
Could you please include an input validation to check that the `artifacts`, `otherArtifacts`, and `hashEquals` slices are the same length before iterating. This will prevent potential panics `index out of range [1] with length 1`

``` golang
artifacts := []*model.ArtifactInputSpec{
    &model.ArtifactInputSpec{Name: "artifact1"},
  }

  otherArtifacts := []*model.ArtifactInputSpec{
    &model.ArtifactInputSpec{Name: "other1"},
    &model.ArtifactInputSpec{Name: "other2"}, 
  }

  hashEquals := []*model.HashEqualInputSpec{
    &model.HashEqualInputSpec{Justification: "same"},
    &model.HashEqualInputSpec{Justification: "same"},
  }
```

Reply:
The input validation has been refactored in https://github.com/guacsec/guac/pull/1210 to be in the resolvers rather than in each backend implementation.
The one for this endpoint is available here https://github.com/guacsec/guac/blob/ebb6442e146d8024ee2bc65eead5f6966215c9d0/pkg/assembler/graphql/resolvers/hashEqual.resolvers.go#L27-L31

Reply:
Thank you! 👍